File: 04.Simulation模拟/leetcode_1848_getMinDistance.cpp
#include <iostream>
#include <vector>
#include <climits>
using namespace std;

class Solution {
public:
    int getMinDistance(vector<int>& nums, int target, int start) {
        int res = INT_MAX;
        for (int i=0; i<nums.size(); i++){
            if (nums[i] == target){
                res = min(res, abs(i-start));
            }
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,2,3,4,5};
    int target = 5, start = 3;
    cout << s.getMinDistance(nums, target, start) << endl;
}

File: 04.Simulation模拟/leetcode_1848_getMinDistance.py
from typing import List

class Solution:
    def getMinDistance(self, nums: List[int], target: int, start: int) -> int:
        return min(abs(start-i) for i in range(len(nums)) if nums[i]==target)

if __name__ == "__main__":
    s = Solution()
    nums = [1,2,3,4,5]
    target = 5
    start = 3
    print(s.getMinDistance(nums, target, start))
